assets/js/public/player/player.js

```diff
@@ -312,6 +319,48 @@ var Player = function(canvas, mpd_list) {
 			gradient.push('rgba(255, 255, 255, 0.3) 100%');
 			element.style.background = 'linear-gradient(' + gradient.join(',') + ')';
 		};
+
+		/* Toggle fullscreen video */
+		this.toggleFullscreen = function() {
+			if (this.fullscreenBtn.className === '') {
+				// player is currently not in full screen mode
+				// set player to full screen
+				var playerArea = document.getElementById('canvasPlayerArea');
+				if (playerArea.webkitRequestFullScreen)
+					playerArea.webkitRequestFullscreen(Element.ALLOW_KEYBOARD_INPUT);   // chrome and safari
+				else if (playerArea.mozRequestFullScreen)
+					playerArea.mozRequestFullScreen();  	// firefox
+				else if (playerArea.msRequestFullScreen)
+					playerArea.msRequestFullScreen();  		// IE
+				else
+					playerArea.requestFullscreen();     	// standard
+				// update button class
+				this.fullscreenBtn.className = 'exit';
+			}
+			else if (this.fullscreenBtn.className == 'exit') {
+				// player is currently in full screen mode
+				// exit from fullscreen
+				if (document.webkitExitFullscreen)
+					document.webkitExitFullscreen();
+				else if (document.mozCancelFullscreen)
+					document.mozCancelFullscreen();
+				else if (document.msExitFullscreen)
+					document.msExitFullscreen();
+				else
+					document.exitFullscreen();
+				// update button class
+				this.fullscreenBtn.className = '';
+			}
+		};
+
+		/* Function to check whether browser is in full screen, called when full screen is toggled */
+		$(document).on('webkitfullscreenchange mozfullscreenchange fullscreenchange MSFullscreenChange', function() {
+			if (!window.screenTop && !window.screenY) {
+				// force update of button class if exiting from full screen
+				// when using esc key and browser button
+				$('#fullscreenBtn').removeClass('exit');
+			}
+		});
```

Comment:
Please check the function above (line 357 onwards) if it is ok with the OOP code. This watch event is needed to correctly update the button on the player if exiting of fullscreen is triggered using escape key or browser button, instead of through the UI.
Also using Jquery (line 361) to update the class for "fullscreenBtn" instead of this.fullscreenBtn.className = ''; as I'm getting undefined when trying to access "this.fullscreenBtn" in this function. Not sure why though :(